drizzle-schema/src/effect-schema.ts

```diff
@@ -0,0 +1,442 @@
+  TRefine extends InsertRefine<TTable>,
+>(
+  table: TTable,
+  refine?: TRefine,
+): BuildInsertSchema<
+  TTable,
+  Equal<TRefine, InsertRefine<TTable>> extends true ? {} : TRefine
+> {
+  const columns = getTableColumns(table);
+  const columnEntries = Object.entries(columns);
+
+  let schemaEntries = Object.fromEntries(
+    columnEntries.map(([name, column]) => {
+      return [name, mapColumnToSchema(column)];
+    }),
+  );
+
+  if (refine) {
+    schemaEntries = Object.assign(
+      schemaEntries,
+      Object.fromEntries(
+        Object.entries(refine).map(([name, refineColumn]) => {
+          return [
+            name,
+            typeof refineColumn === 'function' &&
+              // guard against classes which are also functions, e.g. `Schema.Literal` class
+              !('annotations' in refineColumn)
+              ? refineColumn(schemaEntries as any)
+              : refineColumn,
+          ];
+        }),
+      ),
+    );
+  }
+
+  for (const [name, column] of columnEntries) {
+    if (!column.notNull) {
+      schemaEntries[name] = S.optional(S.NullOr(schemaEntries[name]!)) as any;
+    } else if (column.hasDefault) {
+      schemaEntries[name] = S.optional(schemaEntries[name]!) as any;
+    }
+  }
+
+  return S.Struct(schemaEntries) as any;
+}
+
+// export function createSelectSchema<TTable extends Table>(
+//   table: TTable,
+// ): S.Schema<
+//   Simplify<S.Struct.Constructor<SelectColumnPropertySignatures<TTable>>>
+// > {
+//   const columns = getTableColumns(table);
+//   const columnEntries = Object.entries(columns);
+//
+//   const schemaEntries = Object.fromEntries(
+//     columnEntries.map(([name, column]) => {
+//       return [name, mapColumnToSchema(column)];
+//     }),
+//   );
+//
+//   for (const [name, column] of columnEntries) {
+//     if (!column.notNull) {
+//       schemaEntries[name] = S.NullOr(schemaEntries[name]!);
+//     }
+//   }
+//
+//   return S.Struct(schemaEntries) as any;
+// }
+
+function mapColumnToSchema(column: Column): S.Schema<any, any> {
+  let type: S.Schema<any, any> | undefined;
+
+  if (isWithEnum(column)) {
+    type = column.enumValues.length
+      ? S.Literal(...column.enumValues)
+      : S.String;
+  }
+
+  if (!type) {
+    if (is(column, pgCore.PgUUID)) {
+      type = UUID;
+    } else if (column.dataType === 'custom') {
+      type = S.Any;
+    } else if (column.dataType === 'json') {
+      type = jsonSchema;
+    } else if (column.dataType === 'array') {
+      type = S.Array(
+        mapColumnToSchema((column as PgArray<any, any>).baseColumn),
+      );
+    } else if (column.dataType === 'number') {
+      type = S.Number;
+    } else if (column.dataType === 'bigint') {
+      type = S.BigIntFromSelf;
+    } else if (column.dataType === 'boolean') {
+      type = S.Boolean;
+    } else if (column.dataType === 'date') {
+      type = S.DateFromSelf;
+    } else if (column.dataType === 'string') {
+      let sType = column.notNull ? S.NonEmpty : S.String;
```

Comment:
`""` is `null` for string columns in drizzle?

Reply:
yes I think for sql in general

Reply:
I don't think that empty strings are null in sql, never seen that behavior before

Reply:
guess you're right
<img width="771" alt="image" src="https://github.com/user-attachments/assets/aa3a8b4c-9351-4cdb-a7ce-ae1696c1f3bd">